packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -421,7 +421,9 @@ public void resumeVideoRecording(@NonNull final Result result) {
   }
 
   public void startPreview() throws CameraAccessException {
-    createCaptureSession(CameraDevice.TEMPLATE_PREVIEW, pictureImageReader.getSurface());
+    if (pictureImageReader != null) {
+      createCaptureSession(CameraDevice.TEMPLATE_PREVIEW, pictureImageReader.getSurface());
+    }
```

Comment:
nit: you could invert the if statement to reduce nesting and would make it a bit easier to expand the method:

```suggestion
    if (pictureImageReader == null) {
      return;
    }
    
    createCaptureSession(CameraDevice.TEMPLATE_PREVIEW, pictureImageReader.getSurface());
```

Reply:
Looking further into this issue, it would be good to know why `pictureImageReader` is null at this point and if that is valid (as @tvolkert also mentioned in his comments on PR #2871 ).

If the `pictureImageReader` should not be null this code will hide a mistake that is somewhere else in the code base. I will start looking into this.

Reply:
Thanks Maurits for looking into this! I agree that this is only a workaround for a symptom, but not a proper fix. Do you have a hunch where the real problem could lie?

Reply:
Hi @panmari, we (me and a colleague) have been looking into this issue and can't quite reproduce it as of yet. We do have 2 theories at the moment:

1. Since version 0.5.0 developers are responsible for handling application lifecycle events that might affect the plugin (see also [this comment](https://github.com/flutter/flutter/issues/39109#issuecomment-524250114) mentioned by @kevin-lot). If these events are not handled correctly it might cause the issue where the permission system moves the app to the background and not reinitialize the camera plugin correctly (leaving it in an invalid state);
2. We could did get the same error when (using the example app) very rapidly switching between the front and back facing cameras. We noticed that in these cases the `onOpened` callback would be called when the `dispose` method has already been run (race condition).

I am currently discussing on how we should approach these problems.